tools/generateTool.ps1

```diff
@@ -0,0 +1,121 @@
+
+.LINK
+    The name of a related topic. Repeat this keyword for each related topic.
+
+    This content appears in the Related Links section of the Help topic.
+
+    The Link keyword content can also include a Uniform Resource Identifier
+    (URI) to an online version of the same Help topic. The online version
+    opens when you use the Online parameter of Get-Help. The URI must begin
+    with "http" or "https".
+#>
+
+Param(
+    [Parameter(Mandatory = $true)]
+    [string] $ResourceProvider = "compute/resource-manager",
+    [string] $SpecsRepoFork = "Azure",
+    [string] $SpecsRepoName = "azure-rest-api-specs",
+    [string] $SpecsRepoBranch = "master",
+    [Parameter(Mandatory = $true)]
+    [string] $SdkDirectory= "$([System.IO.Path]::GetTempPath())\Compute",
+    [string] $AutoRestVersion = "latest",
+    [switch] $PowershellInvoker
+)
+
+Write-Host "autorest version received is : $AutoRestVersion"
+
+$errorStream = New-Object -TypeName "System.Text.StringBuilder";
+$outputStream = New-Object -TypeName "System.Text.StringBuilder";
+$currPath = split-path $SCRIPT:MyInvocation.MyCommand.Path -parent
+$modulePath = "$currPath\SdkBuildTools\psModules\CodeGenerationModules\generateDotNetSdkCode.psm1"
+$logFile = "$currPath\..\src\SDKs\_metadata\$($ResourceProvider.Replace("/","_")).txt"
+$errorFile = "$currPath\SdkBuildTools\errorlog.txt"
+
+function NotifyError {
+    param (
+        [string] $errorMsg
+    )
+    Write-Error $errorMsg
+    $errorMsg | Out-File -FilePath $errorFile
+    Start-Process "$errorFile"
+}
+
+if (-not ($modulePath | Test-Path)) {
+    NotifyError "Could not find code generation module at: $modulePath"
+    Exit -1
+}
+
+Import-Module "$modulePath"
+
+if (-not (Get-Module -ListAvailable -Name "$modulePath")) {
+    NotifyError "Could not find module: $modulePath"
+    Exit -1
+}
+
+function Start-Script {
+    Write-InfoLog "Importing code generation module" -logToConsole
+
+    Install-AutoRest $AutoRestVersion
+    
+    $configFile="https://github.com/$SpecsRepoFork/$SpecsRepoName/blob/$SpecsRepoBranch/specification/$ResourceProvider/readme.md"
+    Write-InfoLog "Commencing code generation"  -logToConsole
+    Start-CodeGeneration -SpecsRepoFork $SpecsRepoFork -SpecsRepoBranch $SpecsRepoBranch -SdkDirectory $SdkDirectory -AutoRestVersion $AutoRestVersion -SpecsRepoName $SpecsRepoName
+    
+    $invokerMessage = ".\tools\generate.ps1 was invoked by"
+    if($PowershellInvoker) {
+        $outputStream.Append("$invokerMessage generate.ps1") | Out-Null
+    }
+    else {
+        $outputStream.Append("$invokerMessage generate.cmd") | Out-Null
+    }
+
+}
+
+try {
+    Start-Script
+}
+catch {
+    Write-ErrorLog $_.ToString() -logToConsole
+    Write-ErrorLog $_.ToString() -logToFile
+}
+finally {
+    Get-OutputStream | Out-File -FilePath $logFile | Out-Null
+    Get-ErrorStream | Out-File -FilePath $logFile -Append | Out-Null
+    Clear-OutputStreams
+    Get-Module | Remove-Module
```

Comment:
@dsgouda this will remove all default modules (e.g Management, utility and PSReadLine)
Rather you should only remove module that you imported.